solana/restaking-v2/programs/restaking-v2/src/lib.rs

```diff
@@ -312,6 +372,123 @@ pub mod restaking_v2 {
 
         Ok(())
     }
+
+    pub fn update_token_price(ctx: Context<UpdateTokenPrice>) -> Result<()> {
+        let common_state = &mut ctx.accounts.common_state;
+
+        let token_price_feed = &ctx.accounts.token_price_feed;
+        let sol_price_feed = &ctx.accounts.sol_price_feed;
+
+        let token_mint = ctx.accounts.token_mint.key();
+
+        let validators = common_state.validators.clone();
+
+        let staked_token = common_state
+            .whitelisted_tokens
+            .iter_mut()
+            .find(|whitelisted_token| whitelisted_token.address == token_mint);
```

Comment:
```suggestion
            .find(|whitelisted_token| whitelisted_token.address == token_mint)
            .ok_or_else(|| error!(ErrorCodes::InvalidTokenMint))?;
```

Reply:
as per ur analysis, using lazy evaluation is better.